pandas/core/indexes/base.py

```diff
@@ -5490,8 +5473,22 @@ def insert(self, loc: int, item):
         """
         # Note: this method is overridden by all ExtensionIndex subclasses,
         #  so self is never backed by an EA.
+
+        try:
+            item = self._validate_fill_value(item)
+        except TypeError:
+            if is_scalar(item):
+                dtype, item = maybe_promote(self.dtype, item)
+            else:
```

Comment:
we raise on a scalar? that is odd

Reply:
maybe_promote raises on non-scalar, in particular if we get a tuple here

Reply:
hmm ok can you document in maybe_promote (e.g. add a Raises section).